### PR TITLE
merge master to 5.0.3xx

### PR DESCRIPTION
**Pull request by vlada-shubina**

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -6,8 +6,9 @@
     <add key="dotnet-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" />
     <add key="dotnet5" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json" />
     <add key="dotnet5-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5-transport/nuget/v3/index.json" />
+    <add key="dotnet6" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json" />
```
Comment (bozturkMSFT) [inline NuGet.config:9]: This shouldn't be in 5.0 branch

**Review by bozturkMSFT — CHANGES_REQUESTED** [review]

**Review by bozturkMSFT — APPROVED** [review]

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.5.0/content/ClassLibrary-FSharp/Company.ClassLibrary1.fsproj [hunk 2]
```diff
@@ -5,6 +5,7 @@
     <TargetFramework Condition="'$(TargetFrameworkOverride)' != ''">TargetFrameworkOverride</TargetFramework>
     <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.ClassLibrary1</RootNamespace>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <WarnOn>3390;$(WarnOn)</WarnOn>
```
Comment (DavidKarlas) [inline template_feed/Microsoft.DotNet.Common.ProjectTemplates.5.0/content/ClassLibrary-FSharp/Company.ClassLibrary1.fsproj:8]: I forgot if this change was meant for 5.0.300 or 6.0... Do you remember?

**Review by DavidKarlas — APPROVED** [review]
I added 2 comment about F# WarnOn change

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.5.0/content/ClassLibrary-FSharp/Company.ClassLibrary1.fsproj (hunk 2 above)
Comment (vlada-shubina) [reply template_feed/Microsoft.DotNet.Common.ProjectTemplates.5.0/content/ClassLibrary-FSharp/Company.ClassLibrary1.fsproj:8]: 5.0.300

**Pull request merged**